tmail-backend/guice/distributed/src/test/java/com/linagora/tmail/DockerOpenSearchExtension.java

```diff
@@ -0,0 +1,92 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package com.linagora.tmail;
+
+
+import java.time.Duration;
+import java.util.Optional;
+
+import org.apache.james.GuiceModuleTestExtension;
+import org.apache.james.backends.opensearch.DockerOpenSearch;
+import org.apache.james.backends.opensearch.DockerOpenSearchSingleton;
+import org.apache.james.backends.opensearch.OpenSearchConfiguration;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import com.google.inject.Module;
+
+public class DockerOpenSearchExtension implements GuiceModuleTestExtension {
```

Comment:
Duplicated with the one in test folder of distributed app module?

Can't we just do a little refactor here and reuse this both for distributed and postgres tests?

Reply:
yes, 
I don't want this module to relate to the distributed module, similar to what we did in james-project
Will be refactored later in another separate commit, to avoid conflict in feature when  git rebase